whelktool/scripts/cleanups/2022/10/lxl-3756-move-carriertype.groovy

```diff
@@ -0,0 +1,31 @@
+String where = """
+data['@graph'][1]['hasPart'][0]['carrierType'] IS NOT NULL
+"""
+
+OBJECT_TO_MOVE = ["@id": "https://id.kb.se/marc/OnlineResource"]
+TYPE = "@type"
+ELECTRONIC = "Electronic"
+
+selectBySqlWhere(where) { data ->
+    sourceList = data.graph[1].hasPart[0].carrierType
```

Comment:
```suggestion
    def sourceList = data.graph[1].hasPart[0].carrierType
```
I haven't written Groovy in a while but I believe you need `def` here: "Omitting the "def" keyword puts the variable in the bindings for the current script and groovy treats it (mostly) like a globally scoped variable" - and then Bad Things can happen if whelktool is run multithreaded (default). Been there, done that, got the t-shirt 😅

Reply:
OMG :100: The fact that I missed that might mean that we need to fix the scoping of scripts in the tool itself ("share nothing unless explicitly shared")....